plugins/router/arris-sb6183

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python
+
+# modem:
+#
+# * upstream and downstream power levels
+# * downstream signal to noise ratio
+# * downstream error counts
+#
+# The values are retrieved from the cable modem's status web pages at
+# 192.168.100.1. So, this plugin must be installed on a munin node
+# which can access those pages.
+#
+# To install, place this plugin in the node's plugins directory,
+# /etc/munin/plugins and restart munin-node.
+#
+# Developed and tested with:
+# firmware:         D30CM-OSPREY-2.4.0.1-GA-02-NOSH
+# hardware version: 1
+#
+# Copyright 2020 Nathaniel Clark <nathaniel.clark@misrule.us>
+
+"""
+=head1 NAME
+
+arris-sb6183 - Health monitoring plugin for Arris SB6183 Cable Modem
+
+=head1 CONFIGURATION
+
+Make sure 192.168.100.1 is accessible through your firewall.
+
+To have this register with munin as it's own host set the "env.hostname" in config.
+Also ensure that the hostname set is listed in munin.conf.
+
+[arris*]
+env.hostname modem
+
+=head1 VERSION
+
+0.0.1
+
+=head1 AUTHOR
+
+Nathaniel Clark <nathaniel.clark@misrule.us>
+
+=head1 LICENSE
+
+GPLv2
+
+=head1 MAGIC MARKERS
+
+ #%# family=contrib
+ #%# capabilities=autoconf
+
+=cut
+"""
+
+import re
+import os
+import sys
+import requests
+from lxml import html
```

Comment:
I would suggest to use a standard library instead (e.g. https://docs.python.org/3/library/xml.dom.html). Otherwise you force users to install an additional package and you will need to handle the potentially missing library in the code below (by failing with a reasonable error message in case of an `ImportError`).

Reply:
I would have liked to, but the HTML from the router is poorly structured and xml.etree throws up on parsing it.